### PR TITLE
5227: Build pipeline for Dapi

### PR DESCRIPTION
**Pull request by adrian-fitsios**

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -1,19 +1,72 @@
...
+      azureSubscription: 'NHSAPP-BuyingCatalogue (Non-Prod)'
+      azureContainerRegistry: '{"loginServer":"gpitfuturesdevacr.azurecr.io", "id" : "/subscriptions/7b12a8a2-f06f-456f-b6f9-aa2d92e0b2ec/resourceGroups/gpitfutures-dev-rg-acr/providers/Microsoft.ContainerRegistry/registries/gpitfuturesdevacr"}'
+      dockerComposeFile: '**/docker-compose-build.yml'
+      action: 'Build services'
+      additionalImageTags: '$(Build.BuildNumber)'
+      includeSourceTags: true
+
+- task: DockerCompose@0
+  displayName: 'Push: Dev'
+  inputs:
```
Comment (talawson05) [inline azure-pipelines.yml:61]: you might want a condition to stop pushing to the ACR during a pull request, overwriting the image
something like
`condition: and(succeeded(), not(eq(variables['build.reason'], 'PullRequest')))`

**Comment by talawson05** [comment]
Is there a reason to only build and push to Dev, and not the other containers? Is this a change in methodology?

File: azure-pipelines.yml (hunk 1 above)
Comment (talawson05) [reply azure-pipelines.yml:61]: or use `ne` to trim that second condition down

Comment (adrian-fitsios) [reply azure-pipelines.yml:61]: 👍 Good idea, will do that.

**Comment by adrian-fitsios** [comment]
> Is there a reason to only build and push to Dev, and not the other containers? Is this a change in methodology?

Not that I'm aware of. I will have a chat with @roses1089 and if we actually want to push to other envs I will add that too.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -1,19 +1,124 @@
...
+  displayName: 'Docker Push: Prod'
+  condition: and(succeeded(), ne(variables['build.reason'], 'PullRequest'))
+  inputs:
+    azureSubscription: 'NHSAPP-BuyingCatalogue (Prod)'
+    azureContainerRegistry: '{"loginServer":"gpitfuturesprodacr.azurecr.io", "id" : "/subscriptions/d1be8dbc-1a9f-4b7b-ba51-037116110e00/resourceGroups/gpitfutures-prod-rg-acr/providers/Microsoft.ContainerRegistry/registries/gpitfuturesprodacr"}'
+    dockerComposeFile: '**/docker-compose-build.yml'
+    action: 'Push services'
+
+- script: docker-compose -f "docker-compose.integration.yml" down -v
+  displayName: 'Docker-compose down'
```
Comment (talawson05) [inline azure-pipelines.yml]: Just spotted this, you always want to perform the compose down, even if the build fails. I'd suggest something like
https://docs.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#succeededorfailed
or 
https://docs.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#always

**Review by talawson05 — APPROVED** [review]
LGTM

**Pull request merged**